pkg/cri/resource-manager/policy/builtin/memtier/pod-preferences.go

```diff
@@ -143,6 +147,56 @@ func podSharedCPUPreference(pod cache.Pod, container cache.Container) (bool, int
 	return true, int(elevate)
 }
 
+// ColdStartPreference lists the various ways the container can be configured to trigger
+// cold start. Currently, only timer is supported. If the "duration" is set to a duration
+// greater than 0, cold start is enabled and the DRAM controller is added to the container
+// after the duration has passed.
+type ColdStartPreference struct {
+	duration time.Duration
+}
+
+type coldStartPreferenceYaml struct {
+	Duration string // `yaml:"duration,omitempty"`
+}
+
```

Comment:
Hmm... is this the cleanest way of doing this ? Why not shadow `time.Duration` with `Duration`, define JSON marshallers, and then try unmarshalling directly into a map[string]ColdStartPreference ? Something like this:

```
type ColdStartPreference struct {
	Duration Duration
}
type Duration time.Duration

func (d Duration) MarshalJSON() ([]byte, error) {
	return []byte("\""+time.Duration(d).String()+"\""), nil
}

func (d *Duration) UnmarshalJSON(data []byte) error {
	if len(data) < 2 {
		return fmt.Errorf("invalid Duration data")
	}
	parsed, err := time.ParseDuration(string(data[1:len(data)-1]))
	if err != nil {
		return err
	}
	*d = Duration(parsed)
	return nil
}
```

Reply:
It seems that we would then lose access to `time.Duration` methods, right? I'm seeing error messages such as `type Duration has no field or method String`. Also I think we can't anymore compare `Duration` values with `time.Duration` values without casting, such as checking if the duration is less than one hour. I mean, being able to unmarshall directly makes the code look better, but I'm not sure that it's worth the casting.

Reply:
> It seems that we would then lose access to `time.Duration` methods, right? I'm seeing error messages such as `type Duration has no field or method String`. Also I think we can't anymore compare `Duration` values with `time.Duration` values without casting, such as checking if the duration is less than one hour. I mean, being able to unmarshall directly makes the code look better, but I'm not sure that it's worth the casting.

Hmm... why ? If `ColdStartPreferences` was used internally in many places then I'd understand that comment. But AFAICT on its short journey from `ColdStartPreference.[dD]uration` to `request.coldStart` of type `time.Duration` it is used exactly in one place, in the assignment in `newRequest()`. So the only other change to the rest of the code would be one extra cast there in resource.go:656, where instead of

```
			coldStart = parsedColdStart.duration
```
you would need to have 
```
			coldStart = time.Duration(parsedColdStart.Duration)
```

But let's not spend too much time on this. I'm also fine with the extra type for marshalling. Especially if you know/expect changes that will cause instances of `ColdStartPreference` getting stored/added to some other types as members.